src/psfmachine/machine.py

```diff
@@ -882,56 +902,108 @@ def build_time_model(self, plot=False, downsample=False):
             # Cartesian spline with time dependence
             A3 = _combine_A(A2, time=time_binned)
 
-        # No saturated pixels
-        k = (
-            (flux_binned_raw < 1.4e5).any(axis=0)[None, :]
-            * np.ones(flux_binned_raw.shape, bool)
-        ).ravel()
-        # No faint pixels
-        k &= (
-            (flux_binned_raw > 10).any(axis=0)[None, :]
-            * np.ones(flux_binned_raw.shape, bool)
-        ).ravel()
-        # No huge variability
-        k &= (
-            (np.abs(flux_binned - 1) < 1).all(axis=0)[None, :]
-            * np.ones(flux_binned.shape, bool)
-        ).ravel()
-        # No nans
-        k &= np.isfinite(flux_binned.ravel()) & np.isfinite(flux_err_binned.ravel())
         prior_sigma = np.ones(A3.shape[1]) * 10
         prior_mu = np.zeros(A3.shape[1])
 
-        for count in [0, 1, 2]:
-            sigma_w_inv = A3[k].T.dot(
-                (A3.multiply(1 / flux_err_binned.ravel()[:, None] ** 2)).tocsr()[k]
+        # fit the time model for each segment
+        # we find the splits in data
+        if split_models:
+            dts = np.diff(self.time)
```

Comment:
This line is used in the beginning of the PR, should we have a method for doing this so that it's always the same, and easy to debug? Maybe in utils? Potentially overkill

```python
def get_breaks(time):
    dts = np.diff(self.time)
    return np.hstack([0, np.where(dts > 5 * np.median(dts))[0] + 1, len(self.time)])
```

Reply:
it's an overkill, but definitely easier to debug